private/buf/bufgen/features.go

```diff
@@ -111,9 +111,32 @@ func checkRequiredFeatures(
 				failedFeatures = append(failedFeatures, feature)
 			}
 		}
+		pluginName := configs[responseIndex].Name()
 		if supported&uint64(pluginpb.CodeGeneratorResponse_FEATURE_SUPPORTS_EDITIONS) != 0 && len(required.editionToFilenames) > 0 {
-			// Plugin supports editions, and files include editions. So make sure
-			// the plugin supports precisely the right editions.
+			// Plugin supports editions, and files include editions.
+			// First, let's make sure that the plugin set the min/max edition fields correctly.
+			if response.MinimumEdition == nil {
+				return fmt.Errorf(
+					"plugin %q advertises that it supports editions but did not indicate a minimum supported edition",
+					pluginName,
+				)
+			}
+			if response.MaximumEdition == nil {
+				return fmt.Errorf(
+					"plugin %q advertises that it supports editions but did not indicate a maximum supported edition",
+					pluginName,
+				)
+			}
+			if response.GetMaximumEdition() < response.GetMinimumEdition() {
+				return fmt.Errorf(
```

Comment:
nit: Would we ever want to use `multierr` here to return all of the errors to the user (so they could fix everything wrong w/ the plugin) instead of individually?

Reply:
I don't think so. We only want to validate the actual edition is supported if the response's min and max are valid. And we only want to do this check if both min and max are present. So I don't think they combine in a way that makes a multi error that useful.